### ISSUE TITLE
Syntax highlighting broken

### ISSUE THREAD

**Issue by rodrigoechaide** [hashicorp/vscode-terraform#479]
The following code breaks syntax highlighting. If I remove the `=` sign after `config` syntax highlighting goes back to normal. The same happens if I remove the trailing `"` here `key    = "${var.vpc_tf_state_key}/terraform.tfstate"`. This is happening in version 2.2.0, 2.2.1 and also after rolling back to 2.1.1

```terraform
// Modify vpc, subnets and security group data

data "terraform_remote_state" "vpc" {
  backend = "s3"

  config = {
    bucket = var.backend_bucket
    key    = "${var.vpc_tf_state_key}/terraform.tfstate"
    region = var.backend_region
  }
}

data "aws_ssm_parameter" "dev_db_username" {
  name = "/testoapp/dev_db_username"
}

data "aws_ssm_parameter" "dev_db_password" {
  name = "/testapp/dev_db_password"
}

locals {
  // Remove `Name` tag from the map of tags because Elastic Beanstalk generates the `Name` tag automatically
  // and if it is provided, terraform tries to recreate the application on each `plan/apply`
  // `Namespace` should be removed as well since any string that contains `Name` forces recreation
  // https://github.com/terraform-providers/terraform-provider-aws/issues/3963

  name_prefix = "${var.name}${var.delimiter}${var.environment}${var.delimiter}${var.namespace}"
  tags = merge(
    map("Environment", var.environment),
    map("DeployedBy", "terraform"),
    map("Namespace", var.namespace),
    var.tags
  )
}
```

* Syntax Highlighting Broken

![Capture](https://user-images.githubusercontent.com/14820332/91258231-2d41e980-e76c-11ea-8edc-5f093b758e49.PNG)

* Syntax Highlighting Not Broken (After removing `=` sign right after `config`)

![Capture2](https://user-images.githubusercontent.com/14820332/91258282-4e0a3f00-e76c-11ea-8c03-07b21ceabe59.PNG)

**Comment by surajbarkale-dolby** [comment]
Another broken render because of function call after assignment:

```tf
locals {
  v1 = "1.17"
  m = {
    x = jsondecode(local.v1)

    l = [
      "a", # A (e.g. x, y, z, etc.)
      "b", # B (e.g. PQR)
      "c",
    ]
  }
  v2 = "1.16"
}
```
**Broken**
![broken highlight](https://user-images.githubusercontent.com/5415466/91259733-68670b80-e7b2-11ea-90c9-fa281c1eb982.png)

**Expected with quotes**
![expected highlight](https://user-images.githubusercontent.com/5415466/91259801-8a608e00-e7b2-11ea-99ae-c0f8390e2320.png)

**Comment by freakinhippie** [comment]
I have this issue too, and it seems related to the bare identifiers on the left side of the `=`. If they are quoted the syntax highlighting works again. Obviously it is still a problem, but a workaround can be to quote those identifiers. For example:

```hcl
locals {
  v1 = "1.17"
  ...
}
```

becomes:

```hcl
locals {
  "v1" = "1.17"
  ...
}
```

**Comment by DenWin** [comment]
This seems to be related to #478

**Comment by aeschright** [comment]
Hi folks, we shipped a fix for this in v2.2.2. Please update and file a new issue if you spot additional problems.

**Comment by rodrigoechaide** [comment]
Thanks for the fixes @aeschright but unfortunately after upgrade to v2.2.2 I am still getting the same issue. Should I raise a new issue?
